hawkmoth/parser.py

```diff
@@ -387,6 +397,46 @@ def _clang_diagnostics(diagnostics, errors):
         errors.append(ParserError(ErrorLevel(diag.severity), filename,
                                   diag.location.line, diag.spelling))
 
+def _parse_undocumented_block(domain, comments, errors, cursor, nest):
+    """Parse undocumented blocks.
+
+    Some blocks define plenty of children that may be documented themselves
+    while the parent cursor itself has no documentation. One such example is the
+    `extern "C"` block.
+    """
```

Comment:
> - parser: add mixed domain support
>   - Right now this is only the `extern "C"`, but will be helpful in handling namespaces in C++ as well. I should say I didn't test it yet, but I'm quite confident it will go that way.

And I have spoken too soon of course :/

The issue with namespaces is indeed very similar. But there's an issue of recursion (namespaces within namespaces) that will require a rethink on some of these things. A bit more detail: proper recursion so far implies that there is a docstring on the parent. This commit introduces  a special case to peel off one parent layer that is not documented, but we actually need this inside the recursive parser to be truly generic.

That said, I'd like to have less baggage to rebase as I move forward, so I'd ask for some leniency there if it can be spared. Comments welcome of course, but maybe that could be part of a series to add namespace support.

The plan would go:
1. initial support
2. C++ types support (cpp-class-wip being a preview)
3. namespaces
4. fixes that will surely be needed

Reply:
In fact, I was going to comment that even `extern "C"` and `extern "C++"` can be nested!

And I was going to say it's fine, for now.

The `cursor.hash in comments` will have to move inside `_recursive_parse()` I guess.

Reply:
> In fact, I was going to comment that even extern "C" and extern "C++" can be nested!

Oh, I didn't know that. That's a good test case, but there's no support for it right now of course :/

I wonder if this commit (and the relevant test case) should be left out of the series then. We know this has a solution and this will remain unusable for C++ until we handle namespaces anyway. Fixing this after or when we fix the namespaces will be trivial in principle.

The rest of the series should hold up without it, though I'd have to confirm.